File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        total_product = 1
        for num in nums:
            total_product *= num
        return_lst = []
        for num in nums:
            if num == 0:
                return_lst.append(int(total_product))
            else:
                return_lst.append(int(total_product / num))
        return return_lst

File: Data Structures & Algorithms/two-integer-sum/submission-3.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &nums, int target) {
        unordered_map<int, int> seen;
        int idx = 0;
        for (const int &num : nums) {
            int looking_for = target - num;
            if (seen.find(looking_for) != seen.end()) {
                return vector<int>{seen[looking_for], idx};
            }
            seen[num] = idx;
            idx++;
        }
    }
};
